sqlx-core/src/net/socket/mod.rs

```diff
@@ -182,10 +183,18 @@ impl<S: Socket + ?Sized> Socket for Box<S> {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct TcpKeepalive {
+    pub time: Duration,
+    pub interval: Duration,
+    pub retries: u32,
+}
```

Comment:
https://docs.rs/socket2/latest/socket2/struct.TcpKeepalive.html

I think we should use the builder pattern too and copy the platform support of `socket2` given that this is explicitly done in their case.

Also adding documentation via docstrings is really helpful => copying their docs is likely fine.

Likely we should make this `Copy` too.

Reply:
Thank you for giving me a review!
That's reasonable, or we might have to maintain the consistency between `TcpKeepalive` defined in `socket2` and in `sqlx_core`.
Should I re-export `TcpKeepalive` definition in `socket2`?

Reply:
Current code is a build failour on the following platforms:

```
any(
    target_os = "openbsd",
    target_os = "redox",
    target_os = "solaris",
    target_os = "nto",
    target_os = "espidf",
    target_os = "vita",
    target_os = "haiku",
)
```

Please use the same platform `cfg` scoping as the new dependency.